modules/uhttp/filters.go

```diff
@@ -33,61 +36,83 @@ import (
 // Filter applies filters on requests, request contexts or responses such as
 // adding tracing to the context
 type Filter interface {
-	Apply(ctx fx.Context, w http.ResponseWriter, r *http.Request, next Handler)
+	Apply(ctx context.Context, w http.ResponseWriter, r *http.Request, next Handler)
 }
 
 // FilterFunc is an adaptor to call normal functions to apply filters
-type FilterFunc func(ctx fx.Context, w http.ResponseWriter, r *http.Request, next Handler)
+type FilterFunc func(ctx context.Context, w http.ResponseWriter, r *http.Request, next Handler)
 
 // Apply implements Apply from the Filter interface and simply delegates to the function
-func (f FilterFunc) Apply(ctx fx.Context, w http.ResponseWriter, r *http.Request, next Handler) {
+func (f FilterFunc) Apply(ctx context.Context, w http.ResponseWriter, r *http.Request, next Handler) {
 	f(ctx, w, r, next)
 }
 
-func tracingServerFilter(ctx fx.Context, w http.ResponseWriter, r *http.Request, next Handler) {
-	operationName := r.Method
-	carrier := opentracing.HTTPHeadersCarrier(r.Header)
-	spanCtx, err := ctx.Tracer().Extract(opentracing.HTTPHeaders, carrier)
-	if err != nil && err != opentracing.ErrSpanContextNotFound {
-		ctx.Logger().Info("Malformed inbound tracing context: ", "error", err.Error())
+// contextFilter updates context to fx.Context
+func contextFilter(host service.Host) FilterFunc {
+	return func(ctx context.Context, w http.ResponseWriter, r *http.Request, next Handler) {
+		fxctx := fxcontext.New(ctx, host)
+		next.ServeHTTP(fxctx, w, r)
```

Comment:
IMO this and the other filters should be types that implement `Filter` instead
of using closures to hold the attributes.

```
type contextFilter struct{ Host service.Host }

func (c *contextFilter) Apply(...) {
    fxctx := fxcontext.New(ctx, host)
    next.ServeHTTP(..., w, r)
}
```